File: tests/fixtures/generic/vulnerable/taint_xxe.js
// @audit-fixture
// @rule: taint_xxe
// @expected: detected
const express = require('express');
const app = express();
app.post('/parse', (req, res) => {
    const parser = new DOMParser();
    const doc = parser.parseFromString(req.body.xml, 'text/xml');
    res.json({ ok: true });
});

File: tests/fixtures/generic/vulnerable/taint_xxe.java
// @audit-fixture
// @rule: taint_xxe
// @expected: detected
import javax.servlet.http.HttpServletRequest;
import javax.xml.parsers.*;

public class XmlController {
    public void parse(HttpServletRequest request) throws Exception {
        String xml = request.getParameter("xml");
        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
        factory.newDocumentBuilder().parse(new org.xml.sax.InputSource(new java.io.StringReader(xml)));
    }
}
